### PR TITLE
chore: bump `dasel` and deps versions

### PR DESCRIPTION
**Pull request by hal3e**
Bump `dasel` and dependencies versions. Fixed a `clippy` nightly warning that we should not have functions after a test mod.

### Checklist
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -122,7 +122,7 @@ jobs:
           MIN_VERSION=$(cat Cargo.toml | dasel -r toml 'workspace.package.rust-version')
           RUST_VERSION="${{ env.RUST_VERSION }}"
           echo "Comparing minimum supported toolchain ($MIN_VERSION) with ci toolchain (RUST_VERSION)"
-          test "$MIN_VERSION" == "$RUST_VERSION"
+          test "$MIN_VERSION" == "'$RUST_VERSION'"
```
Comment (Br1ght0ne) [inline .github/workflows/ci.yml:125]:
```suggestion
          test "$MIN_VERSION" == "$RUST_VERSION"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -122,7 +122,7 @@ jobs:
           MIN_VERSION=$(cat Cargo.toml | dasel -r toml 'workspace.package.rust-version')
```
Comment (Br1ght0ne) [inline .github/workflows/ci.yml:122]: `dasel` writes a quoted value to conform to the reader (`toml`) spec. Setting output format to `plain` removes the quotes:

```suggestion
          MIN_VERSION=$(cat Cargo.toml | dasel -r toml 'workspace.package.rust-version' -w plain)
```

**Review by Br1ght0ne — COMMENTED** [review]
LGTM re: deps, a nittest nipick re: `dasel`.

**Review by Br1ght0ne — DISMISSED** [review]

**Review by segfault-magnet — APPROVED** [review]

**Review by Br1ght0ne — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]

**Review by iqdecay — APPROVED** [review]
neat

**Review by digorithm — APPROVED** [review]

**Pull request merged**